.circleci/config.yml

```diff
@@ -0,0 +1,140 @@
+    - run:
+        name: 'opam pin'
+        command: |
+          opam pin add -y reason .
+          opam pin add -y rtop .
+    - save_cache:
+        key: dependency-cache-{{ checksum "../ocaml-version" }}-{{ checksum ".circleci/config.yml" }}-{{ checksum "reason.opam" }}-{{ checksum "rebuild.opam" }}-{{ checksum "rtop.opam" }}
+        paths:
+          - ~/.opam
+          - ~/.npm-global
+    - run:
+        name: 'Test'
+        command: |
+          source ~/.bash_profile
+          eval `opam config env`
+          make test-ci
+    - run:
+        name: 'Check git is clean'
+        command: git diff --exit-code
+
+version: 2
+jobs:
+  4.02.3:
+    docker:
+      - image: ocaml/opam:debian-9_ocaml-4.02.3
+    environment:
+      - TERM: dumb
+      - OCAML_VERSION: "4.02.3"
+      - NPM_CONFIG_PREFIX: "~/.npm-global"
+    <<: *common_steps
+  4.03.0:
+    docker:
+      - image: ocaml/opam:debian-9_ocaml-4.03.0
+    environment:
+      - TERM: dumb
+      - OCAML_VERSION: "4.03.0"
+      - NPM_CONFIG_PREFIX: "~/.npm-global"
+    <<: *common_steps
+  4.04.0:
+    docker:
+      - image: ocaml/opam:debian-9_ocaml-4.04.0
+    environment:
+      - TERM: dumb
+      - OCAML_VERSION: "4.04.0"
+      - NPM_CONFIG_PREFIX: "~/.npm-global"
+    <<: *common_steps
+  4.06.0:
+    docker:
+      - image: ocaml/opam:debian-9_ocaml-4.06.0
+    environment:
+      - TERM: dumb
+      - OCAML_VERSION: "4.06.0"
+      - NPM_CONFIG_PREFIX: "~/.npm-global"
+    <<: *common_steps
+  esy_build:
+    docker:
+      - image: ocaml/opam:debian-9_ocaml-4.06.0
+    environment:
+      - TERM: dumb
+      - OCAML_VERSION: "4.06.0"
+      - NPM_CONFIG_PREFIX: "~/.npm-global"
+    steps:
+      - checkout
+      - run:
+          name: "Install npm"
+          command: |
+            curl -sL https://deb.nodesource.com/setup_9.x | sudo -E bash -
+            sudo apt-get install -y nodejs
+            mkdir -p ~/.npm-global
+            npm config set prefix $NPM_CONFIG_PREFIX
+      - run:
+          name: "Note OCAML_VERSION for cache"
+          command: echo $OCAML_VERSION > ../ocaml-version
+      - restore_cache:
+          key: dependency-cache-{{ checksum "../ocaml-version" }}-{{ checksum ".circleci/config.yml" }}-{{ checksum "reason.opam" }}-{{ checksum "rebuild.opam" }}-{{ checksum "rtop.opam" }}
+      - run:
+          name: "Install esy"
+          command: npm install -g esy@next
+      - run:
+          name: 'Clean'
+          command: make clean-for-ci
+      - run:
+          name: 'opam pin'
+          command: |
+            opam pin add -y reason .
+            opam pin add -y rtop .
+      - run:
+          name: 'esy install'
+          command: ~/.npm-global/bin/esy install
+      - save_cache:
+          key: dependency-cache-{{ checksum "../ocaml-version" }}-{{ checksum ".circleci/config.yml" }}-{{ checksum "reason.opam" }}-{{ checksum "rebuild.opam" }}-{{ checksum "rtop.opam" }}
+          paths:
+            - ~/.opam
```

Comment:
I don't think we need `~/.opam` here either.

Reply:
Removed!